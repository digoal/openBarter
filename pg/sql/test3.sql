SET client_min_messages = warning;
\set ECHO none
\i uninstall_openbarter.sql
\i openbarter.sql
set search_path=market;
\set ECHO all
RESET client_min_messages;
select setval('ob.tdraft_id_seq',100);

select market.fcreate_quality('q1');
select market.fcreate_quality('q2');

select market.fadd_account('o1','postgres>q1',1000);
select market.fadd_account('o2','postgres>q2',1000);

select market.finsert_bid('o1','postgres>q1',100,50,'postgres>q2');
select market.finsert_bid('o2','postgres>q2',100,50,'postgres>q1');
-- executes getdraft_get(6,2,3,2)

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from market.fstats();

select market.faccept_draft(100,'o1');
select market.faccept_draft(100,'o2');

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from market.fstats();

select qown,qname,owner,qtt from market.vowned;
select qown,qname,qtt from market.vbalance;
select id,did,provider,nat,qtt,receiver from market.vmvt;






