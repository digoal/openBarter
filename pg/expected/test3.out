SET client_min_messages = warning;
\set ECHO none
 setval 
--------
      1
(1 row)

RESET client_min_messages;
select setval('ob.tdraft_id_seq',100);
 setval 
--------
    100
(1 row)

select market.fcreate_quality('q1');
 fcreate_quality 
-----------------
               0
(1 row)

select market.fcreate_quality('q2');
 fcreate_quality 
-----------------
               0
(1 row)

select market.fadd_account('o1','postgres>q1',1000);
 fadd_account 
--------------
            0
(1 row)

select market.fadd_account('o2','postgres>q2',1000);
 fadd_account 
--------------
            0
(1 row)

select market.finsert_bid('o1','postgres>q1',100,50,'postgres>q2');
 finsert_bid 
-------------
           0
(1 row)

select market.finsert_bid('o2','postgres>q2',100,50,'postgres>q1');
 finsert_bid 
-------------
           1
(1 row)

-- executes getdraft_get(6,2,3,2)
-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from market.fstats();
 errors 
--------
      0
(1 row)

select market.faccept_draft(100,'o1');
 faccept_draft 
---------------
             0
(1 row)

select market.faccept_draft(100,'o2');
 faccept_draft 
---------------
             0
(1 row)

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from market.fstats();
 errors 
--------
      0
(1 row)

select qown,qname,owner,qtt from market.vowned;
   qown   |    qname    | owner | qtt 
----------+-------------+-------+-----
 postgres | postgres>q2 | o1    | 100
 postgres | postgres>q2 | o2    | 900
 postgres | postgres>q1 | o2    | 100
 postgres | postgres>q1 | o1    | 900
(4 rows)

select qown,qname,qtt from market.vbalance;
   qown   |    qname    | qtt  
----------+-------------+------
 postgres | postgres>q2 | 1000
 postgres | postgres>q1 | 1000
(2 rows)

select id,did,provider,nat,qtt,receiver from market.vmvt;
 id | did | provider |     nat     | qtt  | receiver 
----+-----+----------+-------------+------+----------
  1 |     | market   | postgres>q1 | 1000 | o1
  2 |     | market   | postgres>q2 | 1000 | o2
  3 |   3 | o1       | postgres>q1 |  100 | o2
  4 |   3 | o2       | postgres>q2 |  100 | o1
(4 rows)

